edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/DeviceScopeController.cs

```diff
@@ -0,0 +1,191 @@
+            await this.SendResponse(result);
+        }
+
+        public async Task<EdgeHubScopeResult> HandleDevicesAndModulesInTargetDeviceScope(string actorDeviceId, string actorModuleId, NestedScopeRequest request)
+        {
+            Events.ReceivedScopeRequest(actorDeviceId, request);
+
+            EdgeHubScopeResult result = new EdgeHubScopeResult();
+
+            // Parse the target device ID from the authchain
+            if (!ValidateChainAndGetTargetDeviceId(actorDeviceId, request.AuthChain, out string targetDeviceId))
+            {
+                Events.InvalidAuthchain(request.AuthChain);
+                result.Status = HttpStatusCode.BadRequest;
+                return result;
+            }
+
+            // Actor device is claiming to be our child, and that the target device is its child.
+            // So we should have an authchain already cached for the target device.
+            IEdgeHub edgeHub = await this.edgeHubGetter;
+            Option<string> targetAuthChainOption = await edgeHub.GetAuthChainForIdentity(targetDeviceId);
+            string targetAuthChain = targetAuthChainOption.Expect(() => new UnauthorizedAccessException($"{targetDeviceId} is not a child of this Edge device"));
+
+            // The actor device should be somewhere in the target device's authchain
+            if (!targetAuthChain.Contains(actorDeviceId))
+            {
+                Events.UnauthorizedActor(actorDeviceId, targetDeviceId);
+                result.Status = HttpStatusCode.Unauthorized;
+                return result;
+            }
+
+            // Construct the result
+            IList<ServiceIdentity> identities = await edgeHub.GetDevicesAndModulesInTargetScopeAsync(targetDeviceId);
+            Events.SendingScopeResult(identities);
+            result = MakeResultFromIdentities(identities);
+            result.Status = HttpStatusCode.OK;
+
+            return result;
+        }
+
+        async Task SendResponse(EdgeHubScopeResult result)
+        {
+            this.Response.StatusCode = (int)result.Status;
+
+            if (result.Status == HttpStatusCode.OK)
+            {
+                var resultJsonContent = JsonConvert.SerializeObject(result);
+                var resultUtf8Bytes = Encoding.UTF8.GetBytes(resultJsonContent);
+
+                this.Response.ContentLength = resultUtf8Bytes.Length;
+                this.Response.ContentType = SupportedContentType;
+
+                await this.Response.Body.WriteAsync(resultUtf8Bytes, 0, resultUtf8Bytes.Length);
+            }
+        }
+
+        public static bool ValidateChainAndGetTargetDeviceId(string actorDeviceId, string authChain, out string targetDeviceId)
+        {
+            targetDeviceId = string.Empty;
+            var actorAuthChainIds = authChain.Split(';', StringSplitOptions.RemoveEmptyEntries).ToList();
```

Comment:
nit: why toList?

Reply:
I think I was using something from IList in an earlier iteration, but looks like I don't have it anymore.  Changed back to array.

---
In reply to: [451660023](https://github.com/Azure/iotedge/pull/3149#discussion_r451660023) [](ancestors = 451660023)